app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java

```diff
@@ -164,7 +164,10 @@ else if(noProperRoad == AddRoadNameForm.IS_LINK)
 	@Override public String getCommitMessage() { return "Determine road names and types"; }
 	@Override public int getIcon() { return R.drawable.ic_quest_street_name; }
 	@Override public int getTitle() { return R.string.quest_streetName_title; }
-	@Override public int getTitle(@NonNull Map<String,String> tags) { return getTitle(); }
+	@Override public int getTitle(@NonNull Map<String,String> tags) {
+		if (tags.get("highway").equals("pedestrian")) return R.string.quest_streetName_pedestrian_title;
```

Comment:
possible nullpointer

Reply:
Is it not enough to assume that thanks to https://github.com/ENT8R/StreetComplete/blob/c69465407a868021fb8f843ba9795a2fc47be3b4/app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java#L42 highway tag will be always present?

Reply:
I thought that too...

Reply:
Possibly, there won't be a problem with this now, but there may be in the future:

By contract (=by interface), it is only guaranteed that the `tags` parameter will never be null, it does not say anything about which tags must be present.

Note that the default implementation of `getTitle` in `SimpleOverpassQuestType` is:
```
@Override public final int getTitle()
{
	return getTitle(Collections.emptyMap());
}
```

So, asserting that every call to `getTitle(@NonNull Map<String,String> tags)` must carry the correct `tags` from the associated element with it, comes down to **asserting that `getTitle()` will never be called on `OsmQuestType`s**. Ulteriorly "removing" methods from the interface this way is prone to error, as this is not obvious to anyone.